### PR TITLE
Only to build target platform for cross-compilation by providing goos and goarch.

### PR DESCRIPTION
**Pull request by Kaustubh-pande**
add new **-p** or **--platform** which helps to build target platform for cross-compilation (e.g., linux-amd64, darwin-amd64)"

cc @dsimansk

### CODE REVIEW

File: hack/build.sh [hunk 1]
```diff
@@ -63,6 +63,19 @@ run() {
...
+# platform mode: Only to build target platform for cross-compilation and maybe run test
+  if $(has_flag --platform -p); then
+  # Extract GOOS and GOARCH from command-line arguments
+    GOOS="${ARGS[1]}"
+    GOARCH="${ARGS[2]}"
+    go_build_with_goos_goarch "$GOOS" "$GOARCH"
+
+    if $(has_flag --test -t); then
+       go_test
+    fi
```
Comment (dsimansk) [inline hack/build.sh:75]: What's the reason to keep this if statement inside nested inside if `has_flag`?

File: hack/build.sh [hunk 2]
```diff
@@ -63,6 +63,19 @@ run() {
     exit 0
   fi
 
+# platform mode: Only to build target platform for cross-compilation and maybe run test
+  if $(has_flag --platform -p); then
+  # Extract GOOS and GOARCH from command-line arguments
+    GOOS="${ARGS[1]}"
+    GOARCH="${ARGS[2]}"
```
[the remark this answers is not in the document]
Comment (dsimansk) [reply hack/build.sh:70]: Nevermind I see
```
GOOS ?=
GOARCH ?=
```

File: hack/build.sh [hunk 3]
```diff
@@ -323,6 +355,7 @@ Usage: $(basename $BASH_SOURCE) [... options ...]
 with the following options:
 
 -f  --fast                    Only compile (without dep update, formatting, testing, doc gen)
+-p  --platform                Specify the target platform for cross-compilation (e.g., linux-amd64, darwin-amd64)"
```
Comment (dsimansk) [inline hack/build.sh:358]: Should the example be without dashes? Given the function takes `ARGS[1]` `ARGS[2]`?
```suggestion
-p  --platform                Specify the target platform for cross-compilation (e.g., linux amd64, darwin amd64)"
```

File: hack/build.sh (hunk 1 above)
Comment (Kaustubh-pande) [reply hack/build.sh:75]: I Keep it like other functions, just to be consistent with others.

**Pull request merged**